tools/python/boutiques/bosh.py

```diff
@@ -469,6 +470,85 @@ def pull(*params):
+    if len(params) <= 1 and helps:
+        parser.print_help()
+        raise SystemExit
+
+    args, params = parser.parse_known_args(params)
+    action = args.action
+    params += ["--help"] if args.help is True else []
+
+    if action == "inspect":
+        parser = ArgumentParser("Displays contents of cache")
+        parser.add_argument("-e", "--example", action="store_true",
+                            help="Display example data file contents.")
+        results = parser.parse_args(params)
+
+        from boutiques.dataHandler import DataHandler
+        dataHandler = DataHandler()
+        return dataHandler.inspect(results.example)
+
+    if action == "publish":
+        parser = ArgumentParser("Publishes record(s) to a Zenodo data set.")
+        parser.add_argument("-a", "--author", action="store",
+                            help="Set the author name for the data set "
+                            "publication. Defaults to anonymous.")
+        parser.add_argument("-f", "--file", action="store",
+                            help="Filename of record to publish alone as a "
+                            "data set.")
+        parser.add_argument("-i", "--individually", action="store_true",
+                            help="Publishes all data files in cache as "
+                            "independent data sets, By Default will publish "
+                            "files in bulk data sets.")
+        parser.add_argument("--no-int", '-y', action="store_true",
+                            help="disable interactive input.")
+        parser.add_argument("-v", "--verbose", action="store_true",
+                            help="print information messages.")
+        parser.add_argument("--sandbox", action="store_true",
+                            help="publish to Zenodo's sandbox instead of "
+                            "production server. Recommended for tests.")
+        parser.add_argument("--zenodo-token", action="store",
+                            help="Zenodo API token to use for authentication. "
+                            "If not used, token will be read from "
+                            "configuration file or requested interactively.")
+        results = parser.parse_args(params)
+
+        from boutiques.dataHandler import DataHandler
+        dataHandler = DataHandler()
+        return dataHandler.publish(results.file, results.zenodo_token,
+                                   results.author, results.individually,
+                                   results.sandbox, results.no_int,
+                                   results.verbose)
+
+    if action == "delete":
+        parser = ArgumentParser("Delete data record(s) in cache.")
+        group = parser.add_mutually_exclusive_group(required=True)
+        group.add_argument("-f", "--file", action="store",
+                           help="Filename of record to delete.")
+        group.add_argument("--all", action="store_true",
```

Comment:
--all could be the default behaviour if -f isn't provided.

Reply:
fixed